pkg/kv/kvserver/replica_init.go

```diff
@@ -173,28 +177,26 @@ func (r *Replica) setStartKeyLocked(startKey roachpb.RKey) {
 	r.startKey = startKey
 }
 
-// loadRaftMuLockedReplicaMuLocked will load the state of the replica from disk.
-// If desc is initialized, the Replica will be initialized when this method
-// returns. An initialized Replica may not be reloaded. If this method is called
-// with an uninitialized desc it may be called again later with an initialized
-// desc.
+// loadRaftMuLockedReplicaMuLocked loads the state of the initialized replica
+// from storage. After this method returns, Replica is initialized, and can not
+// be loaded again.
 //
-// This method is called in three places:
+// This method is called in two places:
 //
 //  1. newReplica - used when the store is initializing and during testing
-//  2. tryGetOrCreateReplica - see newUnloadedReplica
-//  3. splitPostApply - this call initializes a previously uninitialized Replica.
+//  2. splitPostApply - this call initializes a previously uninitialized Replica.
 func (r *Replica) loadRaftMuLockedReplicaMuLocked(desc *roachpb.RangeDescriptor) error {
 	ctx := r.AnnotateCtx(context.TODO())
+	if !desc.IsInitialized() {
+		log.Fatalf(ctx, "r%d: cannot load an uninitialized replica", desc.RangeID)
+	}
 	if r.mu.state.Desc != nil && r.IsInitialized() {
```

Comment:
Desc is always non-nil, isn't it? Can't we remove this first part?

Reply:
Done.